syntax = "proto3";

service SceneGRPC {
	rpc GetScene(SceneRequest) returns (SceneResponse){}
}

message Vector3D {
	float x = 1;
	float y = 2;
	float z = 3;
}

message Transform {
	Vector3D position = 1;
	Vector3D rotation = 2;
	Vector3D scale = 3;
}

message Obj {
	string objName = 1;
	bytes objFile = 2;
}

message Model {
	int32 modelID = 1;
	Obj object = 2;
	Transform transform = 3;
}

message SceneResponse {
	string sceneName = 1;
	repeated Model models = 2;
}

message SceneRequest {
	int32 sceneID = 1;
}
